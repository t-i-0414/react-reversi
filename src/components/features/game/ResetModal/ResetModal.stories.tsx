import { ResetModalView } from './ResetModalView';
import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof ResetModalView> = {
  title: 'features/game/ResetModal',
  component: ResetModalView,
  argTypes: {
    resultText: {
      control: {
        type: 'text',
      },
    },
    onResetGame: {
      action: 'onReset',
    },
  },
  args: {
    resultText: 'Draw!',
  },
};

export default meta;

type Story = StoryObj<typeof ResetModalView>;

export const Normal: Story = {
  render: args => <ResetModalView {...args} />,
};
