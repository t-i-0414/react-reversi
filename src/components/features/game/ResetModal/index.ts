export { ResetModalContainer as ResetModal } from './ResetModalContainer';
