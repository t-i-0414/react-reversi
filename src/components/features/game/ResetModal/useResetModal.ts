import { useCallback, useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { GameDomainService, GameResult } from '~/domains';
import { resetGame } from '~/store';
import type { RootState } from '~/store';

export const useResetModal = (): {
  gameResult: GameResult;
  handleResetGame: () => void;
} => {
  const dispatch = useDispatch();
  const game = useSelector((state: RootState) => state.game);

  const gameResult = useMemo(() => GameDomainService.getResult(game), [game]);

  const handleResetGame = useCallback(() => {
    dispatch(resetGame());
  }, [dispatch]);

  return { gameResult, handleResetGame };
};
