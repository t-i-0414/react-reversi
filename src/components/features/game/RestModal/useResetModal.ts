import { useDispatch, useSelector } from 'react-redux';
import { GameDomainService, GameResult } from '~/domains';
import { resetGame } from '~/store';
import type { RootState } from '~/store';

export const useResetModal = (): {
  gameResult: GameResult;
  handleResetGame: () => void;
} => {
  const dispatch = useDispatch();
  const game = useSelector((state: RootState) => state.game);

  const gameResult = GameDomainService.getResult(game);

  const handleResetGame = () => {
    dispatch(resetGame());
  };

  return { gameResult, handleResetGame };
};
