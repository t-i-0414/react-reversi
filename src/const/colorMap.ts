export const ColorMap = {
  // common colors
  WHITE: 'fff',

  // text colors
  TX_DEEP_BLACK: '#000',
  TX_BLACK: '#555',
  TX_WHITE: '#fff',
  TX_RED: '#F62043',

  // background colors
  BG_GREEN: '#38A34B',
  BG_WHITE: '#fff',
  BG_LIGHT_BLACK: '#262626',
  BG_BLACK: '#000',
  BG_BLACK_ALPHA30: 'rgba(0, 0, 0, 0.3)',
  BG_LIGHT_GRAY: '#c7c7c7',

  // border colors
  BD_BLACK: '#000',
  BD_LIGHT_GRAY: '#c7c7c7',

  // piece colors
  PC_WHITE: '#fff',
  PC_BLACK: '#000',
  PC_INVISIBLE: '#38A34B',
} as const;
