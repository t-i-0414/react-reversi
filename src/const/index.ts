export { ColorMap } from '~/const/colorMap';
export { PieceColor } from '~/const/pieceColorMap';
export { PlayerMap } from '~/const/playerMap';
export { SizeMap } from '~/const/sizeMap';
