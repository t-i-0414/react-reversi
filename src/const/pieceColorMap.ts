import type { AllPieceColorMap } from '~/types';

export const PieceColor: AllPieceColorMap = {
  WHITE: 'white',
  BLACK: 'black',
  INVISIBLE: 'invisible',
} as const;
