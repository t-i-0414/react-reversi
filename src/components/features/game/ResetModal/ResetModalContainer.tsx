import React from 'react';
import { ResetModalView } from './ResetModalView';
import { useResetModal } from './useResetModal';

export const ResetModalContainer: React.FC = () => {
  const { handleResetGame, gameResult } = useResetModal();
  const resultText = gameResult.isDraw
    ? 'Draw!'
    : `${gameResult.winner.name} Win!`;

  return (
    <ResetModalView resultText={resultText} onResetGame={handleResetGame} />
  );
};
