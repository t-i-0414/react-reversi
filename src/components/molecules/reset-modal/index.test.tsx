import { expect } from '@jest/globals';
import { render, cleanup } from '@testing-library/react';
import React from 'react';
import useResetModal from './hooks';
import ResetModal from '.';
import { PlayerMap } from '~/const';

jest.mock('./hooks');
const useResetModalMock = useResetModal as jest.MockedFunction<
  typeof useResetModal
>;

describe('resetModal', () => {
  afterEach(cleanup);

  it(`snapshot(${PlayerMap.PLAYER_1.name})`, () => {
    expect.hasAssertions();

    useResetModalMock.mockReturnValue({
      onReset: () => true,
      gameResultText: `${PlayerMap.PLAYER_1.name} win!`,
    });
    const { asFragment } = render(<ResetModal />);
    expect(asFragment()).toMatchSnapshot();
  });

  it(`snapshot(${PlayerMap.PLAYER_2.name})`, () => {
    expect.hasAssertions();

    useResetModalMock.mockReturnValue({
      onReset: () => true,
      gameResultText: `${PlayerMap.PLAYER_2.name} win!`,
    });
    const { asFragment } = render(<ResetModal />);
    expect(asFragment()).toMatchSnapshot();
  });

  it(`snapshot(Draw)`, () => {
    expect.hasAssertions();

    useResetModalMock.mockReturnValue({
      onReset: () => true,
      gameResultText: `Draw!`,
    });
    const { asFragment } = render(<ResetModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
