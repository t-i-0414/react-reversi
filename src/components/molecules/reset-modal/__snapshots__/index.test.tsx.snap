// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetModal snapshot(Draw) 1`] = `
<div
  className="sc-guDLRT eRIqgP"
>
  <div
    className="sc-dmyDGi gSsOMi"
  >
    <p
      className="sc-hLQRIN eNckTq"
    >
      Game Result
    </p>
    <p
      className="sc-eDLJxc ikMfYs"
    >
      Draw!
    </p>
    <button
      className="sc-beySbM MUICB"
      data-cy="reset"
      onClick={[Function]}
      type="submit"
    >
      Game Reset
    </button>
  </div>
</div>
`;

exports[`resetModal snapshot(Player 1) 1`] = `
<div
  className="sc-guDLRT eRIqgP"
>
  <div
    className="sc-dmyDGi gSsOMi"
  >
    <p
      className="sc-hLQRIN eNckTq"
    >
      Game Result
    </p>
    <p
      className="sc-eDLJxc ikMfYs"
    >
      Player 1 win!
    </p>
    <button
      className="sc-beySbM MUICB"
      data-cy="reset"
      onClick={[Function]}
      type="submit"
    >
      Game Reset
    </button>
  </div>
</div>
`;

exports[`resetModal snapshot(Player 2) 1`] = `
<div
  className="sc-guDLRT eRIqgP"
>
  <div
    className="sc-dmyDGi gSsOMi"
  >
    <p
      className="sc-hLQRIN eNckTq"
    >
      Game Result
    </p>
    <p
      className="sc-eDLJxc ikMfYs"
    >
      Player 2 win!
    </p>
    <button
      className="sc-beySbM MUICB"
      data-cy="reset"
      onClick={[Function]}
      type="submit"
    >
      Game Reset
    </button>
  </div>
</div>
`;
